### PR TITLE
Add/update master user on role change

### PR DESCRIPTION
**Pull request by enejb**
#### Changes proposed in this Pull Request:
- Move the role changes to the Jetpack_Sync_Users and we make sure that the class is actually used. 
- Added tests for what happend when the master user isn't an admin any more. 
- Deprecates user_role_change method. 
- And turns some methods static.

### CODE REVIEW

File: sync/class.jetpack-sync-actions.php [hunk 1]
```diff
@@ -74,6 +74,10 @@ static function init() {
 		if ( ! wp_next_scheduled ( 'jetpack_send_db_checksum' ) ) {
 			wp_schedule_event( time(), 'hourly', 'jetpack_send_db_checksum' );
 		}
+
+		// Sync Master User Role Changes
+		require_once dirname( __FILE__ ) . '/class.jetpack-sync-users.php';
```
Comment (gravityrail) [inline sync/class.jetpack-sync-actions.php]: Note that this code will never run if the filter `jetpack_sync_sender_should_load` (above) returns false. Is that your intended behaviour?

I imagine so, but I'm not 100% sure.

Comment (enejb) [inline sync/class.jetpack-sync-actions.php]: Good point. I think I might need to move this somewhere else.

**Comment by gravityrail** [comment]
lgtm! 👍

**Comment by lezama** [comment]
:shipit:

**Pull request merged**